src/utils/validation/registration/contributorValidation.ts

```diff
@@ -5,37 +5,41 @@ import { ErrorMessage } from '../errorMessage';
 
 const contributorValidationSchema = Yup.object().shape({
   correspondingAuthor: Yup.boolean(),
+  sequence: Yup.number(),
+  role: Yup.string(),
   email: Yup.string().when('correspondingAuthor', {
     is: true,
     then: Yup.string().email(ErrorMessage.INVALID_FORMAT).required(ErrorMessage.REQUIRED),
   }),
 });
 
-export const contributorsValidationSchema = Yup.array()
-  .when('$publicationInstanceType', {
-    is: BookType.ANTHOLOGY,
-    then: Yup.array()
-      .of(contributorValidationSchema)
-      .test('editor-test', ErrorMessage.MISSING_EDITOR, (contributors) =>
-        hasRole(contributors, ContributorRole.EDITOR)
-      ),
-    otherwise: Yup.array()
-      .of(contributorValidationSchema)
-      .test('author-test', ErrorMessage.MISSING_AUTHOR, (contributors) =>
-        hasRole(contributors, ContributorRole.CREATOR)
-      ),
-  })
-  .when('$publicationContextType', {
-    is: PublicationType.DEGREE,
-    then: Yup.array()
-      .of(contributorValidationSchema)
-      .test('supervisor-test', ErrorMessage.MISSING_SUPERVISOR, (contributors) =>
-        hasRole(contributors, ContributorRole.SUPERVISOR)
-      )
-      .test('author-test', ErrorMessage.MISSING_AUTHOR, (contributors) =>
-        hasRole(contributors, ContributorRole.CREATOR)
-      ),
-  });
+export const contributorsValidationSchema = Yup.array().when(
+  ['$publicationContextType', '$publicationInstanceType'],
+  (publicationContextType: any, publicationInstanceType: any) => {
```

Comment:
tror egentlig dette er litt shady, for i enkle eksempel ser when slik ut: 
```
.when('$test', (test: string, schema: any) => {
   if (test === "ost") {
      return schema.required();
   } else {
      return schema;
   }
}
```
Hvor `schema` er Yup-reglene som er satt før `.when()` (altså de arves og utvides bare i when)
Men siden vi trenger 2 variabler blir det litt trøblete for da tar plutselig variabel nr 2 plassen til `schema` uten at `schema` kan brukes som argument nr 3 elns(!). Så vi kan faktisk bruke `publicationContextType: string` her, men `publicationInstanceType` må fortsatt være `any` (siden Yup mener at det er `schema`, som det jo vanligvis er i enkle tilfeller med bare en variabel). Syns bare det ble mer forvirrende om disse fikk ulik type, så da satte jeg bare any for begge.
(vet ikke om dette var forståelig, men 🙂)

Docs:
https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema

Reply:
de eksemplene der er faktisk skikkelig dårlige. veldig typisk...